### ISSUE TITLE
Latest publish-please@3.0.1 doesn't install via npm@3.10.10

### ISSUE THREAD

**Issue by Farfurix** [inikulin/publish-please#66]
Hello, we use Node@6 setup in Travis CI check (`nvm install 6`) and have the `publish-please` installation problem.

Full details:
https://travis-ci.org/testcafe-build-bot/testcafe-hammerhead/builds/390074883

Error:
https://travis-ci.org/testcafe-build-bot/testcafe-hammerhead/jobs/390074885#L474
```
module.js:478
    throw err;
    ^
Error: Cannot find module 'chalk'
```

Also it reproduced on Windows 10:
```
D:\workspace\test1>npm -v
3.10.10


D:\workspace\test1>npm install --save-dev publish-please@3.0.1

> publish-please@3.0.1 preinstall D:\workspace\test1\node_modules\.staging\publish-please-e65b8518
> node lib/pre-install.js

internal/modules/cjs/loader.js:596
    throw err;
    ^

Error: Cannot find module 'chalk'
...
```

**Comment by hdorgeval** [comment]
Hi @Farfurix , thank you for your feedback. I will investigate this issue asap. As a work around is it possible to upgrade npm to the latest version in the CI?

**Comment by hdorgeval** [comment]
Hi @Farfurix, it would help me a lot if you could confirm that when launching the pre-install script of publish-please package, there is no node_modules at this time inside the publish-please folder (ie the node_modules only "popup" on post-install)

**Comment by thibaultboursier** [comment]
@hdorgeval Hello, i'm going to try to reproduce this issue.

**Comment by hdorgeval** [comment]
@Farfurix , @thibaultboursier, @miherlosev : impossible to reproduce this error on my side, even when I configure travis and appveyor with node 6 and npm@3.10.10. Maybe there is a hidden difference in the travis configuration between testcafe-hammerhead and publish-please repos.

If think that you are facing this issue when the node_modules folder is not present inside the publish-please installation folder. Maybe it is related to npm cache or to npm default paths.

I hope the workaround of upgrading npm to latest is working for you.

If this issue is critical for you, I can, on my side, modify the pre-install hook in order to fail nicely if the chalk module is not found. Please let me know what is the best for you.

**Comment by thibaultboursier** [comment]
@hdorgeval Same for me, i couldn't reproduce this issue, even with the same NPM/Node/PP version. However, it seems fixed on testcafe side : https://travis-ci.org/testcafe-build-bot/testcafe-hammerhead/builds/391742808

**Comment by Farfurix** [comment]
@hdorgeval, @thibaultboursier: this [PR](https://github.com/DevExpress/testcafe-hammerhead/pull/1644) is closed with the `npm` update workaround. I need some time to provide additional information.

**Comment by hdorgeval** [comment]
Hi @Farfurix, I close this issue. Feel free to re-open it if you encounter the same problem.

**Comment by Farfurix** [comment]
Hello, @hdorgeval.
I confirm that there is no problem with `npm install` in `publish-please` repo context. I'm trying to install `publish-please` as `testcafe-hammerhead` dependency. 
Installation via https://www.katacoda.com/courses/nodejs/playground:
```
  > git clone https://github.com/DevExpress/testcafe-hammerhead.git .

  > npm -v
3.10.10

  > npm install
```
then we get the same error:
```
> publish-please@3.0.2 preinstall /home/scrapbook/tutorial/node_modules/.staging/publish-please-65795ce2
> node lib/pre-install.js

module.js:471
    throw err;
    ^
...
```
This problem can be solved by installing `npm` 5 or higher (now we use workaround with latest `npm` version). In this case (`npm` 5 or higher) we have
`> publish-please@3.0.2 preinstall D:\workspace\hammerhead\node_modules\publish-please`
instead of "staging" path in `npm` 3.10.10:
`> publish-please@3.0.2 preinstall D:\workspace\hammerhead\node_modules\.staging\publish-please-7cef5778`

References:
https://github.com/npm/npm/issues/16706
https://github.com/npm/npm/commit/f4b6a2c
npm 5.0.0 installer changes (https://blog.npmjs.org/post/161081169345/v500):
> Project-level (toplevel) preinstall scripts now run before anything else, and can modify node_modules before the CLI reads it.

**Comment by hdorgeval** [comment]
Hi @Farfurix, thank you for your great feedback. I reopen the issue and will try to fix it asap

**Comment by hdorgeval** [comment]
Hi @Farfurix , the fix has been delivered in version 3.0.3. Thank's for your help on this issue.
